plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/DependencyResolver.java

```diff
@@ -0,0 +1,103 @@
+
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyFilter;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.filter.DependencyFilterUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.ICallable;
+import org.eclipse.m2e.core.embedder.IMavenExecutionContext;
+import org.eclipse.m2e.core.internal.MavenPluginActivator;
+
+public class DependencyResolver {
+
+  /**
+   * Returns all transitive runtime dependencies of the specified Maven artifact
+   * including the artifact itself.
+   * 
+   * @param groupId group ID of the Maven artifact to resolve
+   * @param artifactId artifact ID of the Maven artifact to resolve
+   * @param version version of the Maven artifact to resolve
+   * @return a list of strings in the form groupId:artifactId:version
+   * @throws CoreException if the dependencies could not be resolved
+   */
+  public static List<String> getTransitiveDependencies(
+      String groupId, String artifactId, String version, IProgressMonitor monitor)
+          throws CoreException {
+       
+    final Artifact artifact = new DefaultArtifact(groupId + ":" + artifactId + ":" + version);
+
+    IMavenExecutionContext context = MavenPlugin.getMaven().createExecutionContext();
+    
+    ICallable<List<String>> callable = new ICallable<List<String>>() {
+      @Override
+      public List<String> call(IMavenExecutionContext context, IProgressMonitor monitor)
+          throws CoreException {
+        List<String> dependencies = new ArrayList<>();
+        DependencyFilter filter = DependencyFilterUtils.classpathFilter(JavaScopes.RUNTIME);
+        RepositorySystem system = MavenPluginActivator.getDefault().getRepositorySystem();
+        
+        CollectRequest collectRequest = new CollectRequest();
+        collectRequest.setRoot(new Dependency(artifact, JavaScopes.RUNTIME));
+        collectRequest.setRepositories(centralRepository(system));
+        final DependencyRequest request = new DependencyRequest(collectRequest, filter);
+        RepositorySystemSession session = context.getRepositorySession();
+
+        try {
+          List<ArtifactResult> artifacts =
+              system.resolveDependencies(session, request).getArtifactResults();
```

Comment:
It's worth putting a comment here that NPE seems to indicate corrupt repository.  Which we may want to indicate separately to the user?

Reply:
done